### PR TITLE
[GEOS-10684] WCS 2.0 is ignoring declared bounds and CRS

### PR DESCRIPTION
**Pull request by aaime**
[GEOS-10684](https://osgeo-org.atlassian.net/browse/GEOS-10684)

See ticket for details

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/WCSNetCDFMosaicTest.java [hunk 1]
```diff
@@ -205,9 +204,9 @@ private void setupNetCDFoutSettings(QName name) {
     private void setupNetCDFoutSettings(QName name, boolean setNoData) {
         CoverageInfo info = getCatalog().getCoverageByName(getLayerId(name));
 
-        // Set the Declared SRS
-        info.setSRS("EPSG:4326");
-        info.setProjectionPolicy(ProjectionPolicy.REPROJECT_TO_DECLARED);
+        //        // Set the Declared SRS
```
Comment (dromagnoli) [inline src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/WCSNetCDFMosaicTest.java:207]: Are these comments intentionally left here?

File: src/wcs2_0/src/main/java/org/geoserver/wcs2_0/GetCoverage.java [hunk 2]
```diff
@@ -1520,9 +1527,11 @@ private CoordinateReferenceSystem extractCRSInternal(
      * @return the subsettingCRS as a {@link CoordinateReferenceSystem}
      */
     private CoordinateReferenceSystem extractSubsettingCRS(
-            GridCoverage2DReader reader, Map<String, ExtensionItemType> extensions) {
+            CoverageInfo ci,
+            GridCoverage2DReader reader,
```
Comment (dromagnoli) [inline src/wcs2_0/src/main/java/org/geoserver/wcs2_0/GetCoverage.java:1531]: Is the reader argument still needed?

File: src/wcs2_0/src/main/java/org/geoserver/wcs2_0/GetCoverage.java [hunk 3]
```diff
@@ -1596,10 +1605,12 @@ private Map<String, ExtensionItemType> extractExtensions(GetCoverageType request
 
     /** */
     private Map<String, InterpolationPolicy> extractInterpolation(
-            GridCoverage2DReader reader, Map<String, ExtensionItemType> extensions) {
+            CoverageInfo ci,
+            GridCoverage2DReader reader,
```
Comment (dromagnoli) [inline src/wcs2_0/src/main/java/org/geoserver/wcs2_0/GetCoverage.java:1609]: Is the reader argument still needed?

File: src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java [hunk 4]
```diff
@@ -252,6 +254,38 @@ protected void onSetUp(SystemTestData testData) throws Exception {
         namespaces.put("wcsgs", "http://www.geoserver.org/wcsgs/2.0");
         XMLUnit.setXpathNamespaceContext(new SimpleNamespaceContext(namespaces));
         xpath = XMLUnit.newXpathEngine();
+
+        // alter the native BBOX of one raster for at least a pixel to make sure the declared
+        // bounds are used, but also fitted to the grid to avoid resampling.
+        // Compared to the native bbox we are expanding by almost one pixel in all directions:
+        //          1 pixel
+        //             ↓
```
Comment (dromagnoli) [inline src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java:262]:
```suggestion
        //             ↑
```

File: src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java [hunk 5]
```diff
@@ -252,6 +254,38 @@ protected void onSetUp(SystemTestData testData) throws Exception {
...
+        //  1 pixel ←-   -→ 1 pixel
+        //             ↑
```
Comment (dromagnoli) [inline src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java:264]:
```suggestion
        //             ↓
```

File: src/wcs2_0/src/test/java/org/geoserver/wcs2_0/kvp/GetCapabilitiesTest.java [hunk 6]
```diff
@@ -43,7 +43,7 @@ public void cleanupLimitedSRS() {
     @Test
     public void testBasicKVP() throws Exception {
         Document dom = getAsDOM("wcs?request=GetCapabilities&service=WCS");
-        // print(dom);
+        print(dom);
```
Comment (dromagnoli) [inline src/wcs2_0/src/test/java/org/geoserver/wcs2_0/kvp/GetCapabilitiesTest.java:46]: Should that be commented again after development?

**Review by dromagnoli — APPROVED** [review]
Looks good to me.
Just reported a few questions on some arguments not needed anymore as well as some commented/uncommented sections here and there.

**Comment by bradh** [comment]
Restarted the JDK 8 build - fail looked unrelated.

File: src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/WCSNetCDFMosaicTest.java (hunk 1 above)
Comment (aaime) [reply src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/WCSNetCDFMosaicTest.java:207]: Nope, I should have removed the lines, doing so.

File: src/wcs2_0/src/main/java/org/geoserver/wcs2_0/GetCoverage.java (hunk 2 above)
Comment (aaime) [reply src/wcs2_0/src/main/java/org/geoserver/wcs2_0/GetCoverage.java:1531]: Nope, good catch.

File: src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java (hunk 4 above)
Comment (aaime) [reply src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java:262]: Arrows is correct, but comment above is inconsistent, fixing.

File: src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java (hunk 5 above)
Comment (aaime) [reply src/wcs2_0/src/test/java/org/geoserver/wcs2_0/WCSTestSupport.java:264]: Same as above

**Comment by aaime** [comment]
@dromagnoli feedback addressed.

**Comment by aaime** [comment]
Merged on main. I'm going to hold backports as changes are a bit invasive for a stable series.

**Pull request merged**